libbeat/processors/add_process_metadata/config.go

```diff
@@ -0,0 +1,134 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package add_process_metadata
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common"
+)
+
+type config struct {
+	// IgnoreMissing: Ignore errors if event has no PID field.
+	IgnoreMissing bool `config:"ignore_missing"`
+
+	// OverwriteFields allow target_fields to overwrite existing fields.
+	OverwriteFields bool `config:"overwrite_fields"`
+
+	// RestrictedFields make restricted fields available (i.e. env).
+	RestrictedFields bool `config:"restricted_fields"`
+
+	// IgnoreErrors causes the event not to be discarded when errors
+	// are encountered during enrichment.
+	// For example when a PID field cannot be converted to an integer or
+	// a field cannot be overwritten. It is still an error if no PID field
+	// is found. Use `ignore_missing` for that.
+	// Instead of the event being dropped, it will return the original event.
+	IgnoreErrors bool `config:"ignore_errors"`
+
+	// MatchPIDs fields containing the PID to lookup.
+	MatchPIDs []string `config:"match_pids"`
+
+	// Target is the destination root where fields will be added.
+	Target string `config:"target"`
+
+	// Fields is the list of fields to add to target.
+	Fields []string `config:"fields"`
+}
+
+// available fields by default
+var validFields = common.MapStr{
```

Comment:
nit: call it availableFields as in the description? valid sounds like the restricted fields would be invalid.

Reply:
I renamed it to defaultFields, which makes more sense.